### DISCUSSION TITLE
NPM Error while installing Sanity Studio v3

### DISCUSSION THREAD

**Discussion by jozefini** [sanity-io/sanity#3361] [Sanity Studio Support] [answered]
Hello, I'm having trouble installing Sanity v3.

`npm create sanity@dev-preview` Throws an error to me.

Here is the full log.

```
PS C:\Users\User\Desktop\studio> npm create sanity@dev-preview
node:events:515
      throw er; // Unhandled 'error' event
      ^

Error: spawn C:\Users\User\AppData\Local\Yarn\Data\global\node_modules\@sanity\cli\bin\sanity ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:289:12)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn C:\\Users\\User\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\@sanity\\cli\\bin\\sanity',
  path: 'C:\\Users\\User\\AppData\\Local\\Yarn\\Data\\global\\node_modules\\@sanity\\cli\\bin\\sanity',
  spawnargs: [ 'init' ]
}

Node.js v18.4.0
npm ERR! code 1
npm ERR! path C:\Users\User\Desktop\studio
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c create-sanity

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\User\AppData\Local\npm-cache\_logs\2022-06-19T10_20_00_821Z-debug-0.log
PS C:\Users\User\Desktop\studio>
```

Node versions: 18.4.0
NPM version: 8.12.1
OS: Windows 10

**Comment by rexxars** [comment]
Could you try running `npx @sanity/cli@dev-preview init` instead?

**Reply by jozefini** [reply]
That command works great however, the Sanity version is still 2 `@sanity/base": "^2.30.1"`

**Reply by jozefini** [reply]
I wonder why is Yarn there `Error: spawn C:\Users\User\AppData\Local\Yarn\Data\global\node_modules\@sanity\cli\bin\sanity`
Seems like is looking at Yarn packages or idk, I don't know anything about those package installers.

**Reply by olehhe** [reply]
Probably have to do `npm install @sanity/cli@dev-preview` first, then `npx @sanity/cli@dev-preview --version` should show something like `@sanity/cli version 3.0.0-dev-preview.9`. 
Then you can try and run `npx @sanity/cli@dev-preview init`.

Worked for me atleast!

**Comment by jozefini** [comment, marked as answer]
Thanks for the answers. The working solution was by @olehhe.

Steps:
1. `npm install @sanity/cli@dev-preview`
2. `npx @sanity/cli@dev-preview --version`
3. `npx @sanity/cli@dev-preview init`
4. Choose `npm` as runtime over `yarn` on Windows OS.

I noticed that choosing Yarn runtime in a windows environment triggers a cli error for the `basepath` when I run `yarn start`. 

Credits to @olehhe again for helping me.
